core/client/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -157,7 +166,7 @@ public void close() throws IOException {
       return;
     }
     updateStreams();
-    if (mCurrentCacheStream != null && mShouldCachePartiallyReadBlock) {
+    if (mShouldCachePartiallyReadBlock) {
```

Comment:
mCurrentCacheStream is not required to readCurrentBlockToEnd(). It was checked before for optimization.